### ISSUE TITLE
SUPPORT REQUEST FOR MSI GS73 8RF BATTERY MASTER

### ISSUE THREAD

**Issue by escapeebay** [BeardOverflow/msi-ec#182]
### Laptop model

GS73 8RF

### EC firmware version

17B7EMS1.103

### EC memory dump

```text
00=00 01=80 02=00 03=00 04=00 05=00 06=00 07=00 
08=00 09=00 0A=00 0B=00 0C=00 0D=00 0E=00 0F=00 
10=00 11=00 12=00 13=00 14=00 15=00 16=00 17=00 
18=00 19=00 1A=00 1B=00 1C=00 1D=00 1E=00 1F=00 
20=00 21=00 22=00 23=00 24=00 25=00 26=00 27=00 
28=00 29=00 2A=00 2B=B0 2C=E0 2D=06 2E=09 2F=0B 
30=03 31=09 32=01 33=0D 34=50 35=0A 36=05 37=00 
38=CC 39=10 3A=88 3B=2C 3C=AE 3D=01 3E=E0 3F=00 
40=00 41=00 42=64 43=00 44=C2 45=0E 46=00 47=00 
48=C2 49=0E 4A=1D 4B=32 4C=EF 4D=0B 4E=00 4F=00 
50=00 51=00 52=00 53=00 54=00 55=00 56=00 57=00 
58=00 59=00 5A=00 5B=00 5C=00 5D=00 5E=00 5F=00 
60=00 61=00 62=00 63=00 64=00 65=00 66=00 67=00 
68=50 69=00 6A=37 6B=3D 6C=43 6D=49 6E=4F 6F=55 
70=64 71=46 72=1E 73=2D 74=37 75=41 76=46 77=46 
78=46 79=00 7A=08 7B=03 7C=03 7D=03 7E=03 7F=03 
80=00 81=00 82=37 83=3D 84=43 85=49 86=4F 87=57 
88=61 89=00 8A=00 8B=32 8C=3C 8D=46 8E=46 8F=46 
90=50 91=00 92=08 93=03 94=03 95=03 96=03 97=02 
98=02 99=00 9A=64 9B=02 9C=00 9D=6E 9E=00 9F=00 
A0=31 A1=37 A2=42 A3=37 A4=45 A5=4D A6=53 A7=31 
A8=2E A9=31 AA=30 AB=33 AC=30 AD=34 AE=32 AF=30 
B0=32 B1=30 B2=31 B3=38 B4=31 B5=30 B6=3A B7=35 
B8=33 B9=3A BA=35 BB=39 BC=C2 BD=0E BE=40 BF=10 
C0=00 C1=01 C2=24 C3=00 C4=00 C5=9E C6=C0 C7=00 
C8=00 C9=DD CA=01 CB=0A CC=00 CD=89 CE=14 CF=00 
D0=00 D1=00 D2=00 D3=00 D4=70 D5=00 D6=00 D7=00 
D8=00 D9=00 DA=00 DB=80 DC=81 DD=00 DE=00 DF=00 
E0=E2 E1=02 E2=00 E3=00 E4=10 E5=00 E6=00 E7=00 
E8=00 E9=00 EA=00 EB=00 EC=00 ED=D1 EE=00 EF=D0 
F0=00 F1=00 F2=C0 F3=00 F4=0C F5=00 F6=03 F7=00 
F8=00 F9=C3 FA=32 FB=00 FC=00 FD=00 FE=00 FF=00
```

### GPU

Nvidia

### Is your keyboard RGB?

Yes (multi color RGB)

### Additional context

I REQUEST BATTERY MASTER SUPPORT FOR GS73 8RF. EC TEST SOFTWARE WAS PREVIOUSLY MADE FOR THE GS65 8RF MODEL, WHICH WAS PRODUCED IN THE SAME YEAR. EC VERSION APPEALS AS 16Q2EMS1.T40. IN MY NET SEARCH, YOU CAN SET THE BATTERY LIMIT TO 100, 80 OR 60 BY CHANGING THE 'EF' ORDER. FOR 100, EF CODE IS E4, FOR 80, EF CODE IS D0 AND FOR 60, EF CODE IS BC. I AM MAKING THIS CHANGE THROUGH DRAGON CENTER. THESE CODES ARE CHANGING VIA EC. BUT IT DOES NOT LIMIT THE CHARGING BECAUSE THE EC SOFTWARE DOES NOT HAVE THIS FEATURE. ATTACHMENT ARE ALL FILES AVAILABLE
[Yeni klasör.zip](https://github.com/user-attachments/files/18270970/Yeni.klasor.zip)

**Comment by glpnk** [comment]
16q2EMS1 file is different device `GS65 Stealth Thin 8R`

**Comment by escapeebay** [comment]
Yes, I know a different device. I am asking if we can do the same work for gs73 8rf. I do not have any coding knowledge.

**Comment by escapeebay** [comment]
[17B7EMS1.103.zip](https://github.com/user-attachments/files/18272240/17B7EMS1.103.zip)
The original and the version I used are attached.

**Comment by escapeebay** [comment]
just open RWEverything, press the EC button (blue box) and write to the register EF (green box).

BC means 60% charge

D0 means 80% charge

E4 means full charge

Battery master menu opened in dragon center.Since there is no such feature in the software, it does not cut the charge at the value I set. Battery master needs to be added to the EC software and optimized
![80-per-cent](https://github.com/user-attachments/assets/f6d2b7c6-7124-4a78-a6fe-2f42b12a67d2)

**Comment by glpnk** [comment]
> Battery master menu opened in dragon center. <....> Battery master needs to be added to the EC software and optimized

What do you mean exactly? How this Battery Master looks like?

> just open RWEverything, press the EC button (blue box) and write to the register EF (green box).
>
> BC means 60% charge
>
> D0 means 80% charge
>
> E4 means full charge

Default address and values for older devices

**Comment by escapeebay** [comment]
My device is already old. Intel 8th generation

**Comment by escapeebay** [comment]
Battery master is not active or coded in my installed ec software. A software developer friend modified the ec software for the gs65 8rf model and added battery master and battery optimization. and named the ec software test. I need to open the ec software and add the battery master codes to it.

**Comment by glpnk** [comment]
Which EC software?

**Comment by escapeebay** [comment]
The ec software I used is 17B7EMS1.103. This is the latest released version. The latest EC version published on the msi support page of the Gs65 8rf model is 16Q2EMS1.107. The EC version was updated again by a software developer friend and named 16Q2EMS1.T40 EC TR.
![gs73 8rf ec msi support](https://github.com/user-attachments/assets/d120d228-9a47-4e1f-b9f4-933acfa9535f)
![gs65 8rf ec msi support](https://github.com/user-attachments/assets/030f8260-7ff8-44e9-8efd-e053a47aafa8)
![I think the version shared by github is](https://github.com/user-attachments/assets/7856a044-28c1-4f4d-88b4-8e1ef8daa408)
![continues to charge  80% selected should not be charged](https://github.com/user-attachments/assets/fe6dd5c0-8406-49a8-b216-82512a869d70)

**Comment by escapeebay** [comment]
I share the original EC software and test software. For friends who want to add code.  Version 17B7EMS1.103 needs to be refactored.
[Uploading ec files.zip…]()

**Comment by escapeebay** [comment]
I shared the gs65 8rf ec software for reference.  Please help refactor gs73 8rf firmware for battery master.

**Comment by escapeebay** [comment]
> Github dosyanızı almadan önce yorum gönderdiniz

Which file?

**Comment by glpnk** [comment]
How? It's written for Intel 8051-like microcontroller, without any sources and built with closed source compiler.

If you mean computer program or driver, and your device support battery charge limiting by hardware, sure, but this project is intended only for Linux.

> Which file?

In this comment https://github.com/BeardOverflow/msi-ec/issues/182#issuecomment-2565705595

**Comment by escapeebay** [comment]
I don't use Linux. Does the motherboard support charging cut-off?
 I am thinking  . I don't know how to recompile my ec file

**Comment by escapeebay** [comment]
Can't we open the ec file, add the codes we want and close it again?

**Comment by glpnk** [comment]
It's not a file, but registers with settings. So you can't open it as file.

So you have "Dragon Center" MSI app, which have a menu for setting charge limit, but it doesn't work? But changing value through "RW Everything" on your screenshot https://github.com/BeardOverflow/msi-ec/issues/182#issuecomment-2565355437 works?

If both answers are yes - than try to reinstall MSI app, it probably broken.

Does this worked in the past?

**Comment by glpnk** [comment]
For removing all MSI apps, use their Cleaning App https://download.msi.com/uti_exe/nb/CleanCenterMaster.zip

**Comment by escapeebay** [comment]
> It's not a file, but registers with settings. So you can't open it as file.
> 
> So you have "Dragon Center" MSI app, which have a menu for setting charge limit, but it doesn't work? But changing value through "RW Everything" on your screenshot [#182 (comment)](https://github.com/BeardOverflow/msi-ec/issues/182#issuecomment-2565355437) works?
> 
> If both answers are yes - than try to reinstall MSI app, it probably broken.
> 
> Does this worked in the past?

No it doesn't work. Because this feature is not added in the current EC software. I think those lines of code are empty and have no effect. Because even if I write those codes manually, they do not affect the charging cut-off.

**Comment by escapeebay** [comment]
https://www.reddit.com/r/MSILaptops/comments/jf0bwx/adjusting_battery_charge_level_without_dragon/?rdt=64549 
fully explained here

**Comment by escapeebay** [comment]
![alıntı](https://github.com/user-attachments/assets/0349d410-3e8d-43d7-8186-37971e779f1d)

**Comment by escapeebay** [comment]
Who made this testing software? MSI didn't do it because I asked, they said it wasn't a version we released.

**Comment by escapeebay** [comment]
Which engineer or software developer friend might have compiled it? Maybe if I contact him, he can compile the EC software for my notebook model.

**Comment by escapeebay** [comment]
reverse engineering

**Comment by glpnk** [comment]
Okay, if hardware is not supports it, than it's may be impossible to tune battery charge limit.

Different devices may use different charging controller, and writing wrong firmware may break your laptop. EC controls keyboard, so writing different firmware will break keyboard

**Comment by escapeebay** [comment]
I am aware of the risk. Maybe it has been studied before and results have been obtained.

**Comment by glpnk** [comment]
No, it's closed source software, probably made to order. Your device is already old, and warranty time is probably expired, so MSI will not support you.

Compiler for 8051 architecture produce highly optimized code, so I doubt that merging 2 firmwares is possible

**Comment by glpnk** [comment]
You can ask someone on the MSI forum, but it will probably not be free.

**Comment by escapeebay** [comment]
I understand thanks a lot. You say Intel 8051 is difficult. Such a study might have been done by a friend. I sent a message to the author himself. maybe he will come back

**Comment by escapeebay** [comment]
![laptop-ec-board](https://github.com/user-attachments/assets/5b3e0400-bd7e-4488-a0c4-20af16b19a72)

**Comment by escapeebay** [comment]
https://hackaday.com/2022/06/07/ec-hacking-your-laptop-has-a-microcontroller/

**Comment by glpnk** [comment]
Good luck, you probably can buy schematics on your laptop, Keil compiller, but not sure about sources for EC
